src/Compiler/Facilities/BuildGraph.fs

```diff
@@ -125,20 +127,23 @@ type NodeCode private () =
     static member RunImmediate(computation: NodeCode<'T>, ct: CancellationToken) =
         let diagnosticsLogger = DiagnosticsThreadStatics.DiagnosticsLogger
         let phase = DiagnosticsThreadStatics.BuildPhase
+        let ct2 = Cancellable.Token
 
         try
             try
                 let work =
                     async {
                         DiagnosticsThreadStatics.DiagnosticsLogger <- diagnosticsLogger
                         DiagnosticsThreadStatics.BuildPhase <- phase
+                        Cancellable.Token <- ct2
```

Comment:
I thought about it too. Unfortunately, even though these two tokens are normally expected to be the same, they live independently now. I decided to maintain the thread-static one here for now, so even if there's any case where the tokens are different, it would still keep the one set via `UsingToken` in the top level code.